cmd/rotate_certs.go

```diff
@@ -148,8 +157,22 @@ func (rcc *rotateCertsCmd) run(cmd *cobra.Command, args []string) error {
 
 	log.Infoln("Generating new certificates")
 
-	// reset the certificateProfile and use the exisiting certificate generation code to generate new certificates.
-	rcc.containerService.Properties.CertificateProfile = &api.CertificateProfile{}
+	// reset the certificateProfile and use the existing certificate generation code to generate new certificates.
+	ca := ""
+	caKey := ""
+
+	if err = validatePerm(rcc.containerService.Properties.CertificateProfile.CaCertificate); err == nil {
```

Comment:
Is it OK to continue if `validatePerm` returns an error?

Reply:
If it returns an error then an empty string will be used. This would mean the CA Certificates will be regenerated. Rotating the Ca certificates is not supported at this time so we may need to fail fast if the root certificates cannot be validated

Reply:
Please make sure documentation explains if it does not, or at least log a message to stdout